### DISCUSSION TITLE
Extend scaffolding beyond their normal limit of 7

### DISCUSSION THREAD

**Discussion by emielderckx** [PurpurMC/Purpur#824] [__Dismissed__] [answered]
Make the scaffolding limit configurable, so it allows you to configure how far scaffolding can reach without support

**Comment by BillyGalbreath** [comment, marked as answer]
Unfortunately, this value is stored in the blockstate with a hardcoded maximum of 7. This means if we set it any higher than that the client will crash.

**Reply by emielderckx** [reply]
https://modrinth.com/mod/scaffolding-extension saw this mod do it server side

**Reply by BillyGalbreath** [reply]
Indeed, but they're able to better control blockstates in a modded environment designed for that kind of thing. Looking at their code I can see where they [inject a new property](https://gitlab.com/supersaiyansubtlety/scaffolding_extension/-/blob/master/src/main/java/net/sssubtlety/scaffolding_extension/ScaffoldingExtension.java#L27) to the blockstate, then are able to [reorder the blockstate palette](https://gitlab.com/supersaiyansubtlety/scaffolding_extension/-/blob/master/src/main/java/net/sssubtlety/scaffolding_extension/ScaffoldingExtension.java#L28) (I'm assuming to put the custom properties at the end of everything vanilla so it doesnt crash clients). they're able to do this with a [specialized lib](https://github.com/Virtuoel/Statement) for modded servers. We have no such luxuries in Bukkit land, nor does anyone (including myself) have the time or know-how to build such a system.
